autotest.pm

```diff
@@ -149,6 +202,33 @@ sub loadtest ($script, %args) {
             EOM
         $is_python = 1;
     }
+    elsif ($script =~ m/\.lua$/) {
+        _load_lua();    # this loads (actually defines) lua_eval()
+        my $script_dir = path($script_path)->dirname->to_abs;
+        $code .= sprintf <<~'EOM', $script_dir, $script_path;
+            use base 'basetest';
+
+            use autotest;
+            autotest::_load_lua();
+            autotest::lua_append_path("%s/../lib/?.lua");
+            autotest::lua_eval('dofile("%s")');
+
+            sub run($self) {
+                # not passing $self as blessed objects generate a warning
```

Comment:
I can imagine a workaround for passing the basetest object.
And since the basetest is practically a singleton at the time of the test run, we could provide a proxy class in lua which calls perl functions to access the original perl object, stored in a global variable.